brainwidemap/decoding/README.md

```diff
@@ -0,0 +1,124 @@
+across trials and targets are task or behavior variables across trials.  Neurons within a given session
+and region are used for a single decoding.  In order to analyze all of the BWM regions we parallelize the 
+code on a slurm based cluster.  The following pipeline steps load the data, perform the regression (both for 
+the real data and for a non-parametric null distribution), and save the results in a format which is 
+accessible for plotting.
+
+You must save a 'settings.py' file in this directory before executing this pipeline.  The file needs to be 
+in the format of settings_template.py.  You may copy and rename that file and make the necessary settings 
+changes.  This file also determines where files will be saved and the file namings for this decoding run.  
+Note that the date is important for file naming, and if you change the date or any other settings related 
+to the variable 'SETTINGS_FORMAT_NAME', then the decoding is treated as a different run. 
+
+For an example of how to perform decoding for a single session, see the file 
+`decoding_example_script.py` in this directory. The instructions below detail how to run a
+large-scale decoding analysis across the entire brainwide map. 
+
+## Caching the data
+
+We do not want to load data from ONE remotely, so we first cache all the data we need.  Subsequent pipeline calls 
+to ONE can then be done in the 'local' mode.
+
+Run the 00_data_caching.py file to do this.  Caching takes many hours, and so should be done in parallel.  We
+submit 50 jobs (see N_PARA and para_index in script) using 
+
+```
+sbatch slurm_data_caching.py
+```
+
+## Caching imposter sessions
+
+Imposter sessions are used to provide a set of pseudo regression targets.  These pseudo regression targets are
+decoded to produce a null distribution for our decoding test statistics.  This step caches these targets so
+that they can be accessed locally as in the previous step.  The imposter sessions may be taken from ephys 
+choice world or biased choice world.  This is determined by the settings.py file parameter, i.e. whether  
+'imposter_generate_from_ephys' is True of False, respectively. 
+The 01_imposter_caching.py script does this and is run in parallel (default of 500 jobs) like the last step 
+with the command:
+
+```
+sbatch slurm_imposter_caching.sh
+```
+
+## Generate imposter dataframe
+
+The imposter sessions from the last step need to be aggregated into a pandas dataframe to be easily accessed 
+by the rest of the pipeline.  This is done by 02_generate_imposter_df.py.  This is not done is parallel, but 
+nevertheless, can be summited to a slurm cluster using
+
+```
+sbatch slurm_generate_imposter_df.py
+```
+
+## Decoding sessions
+
+The regression is performed in this step by running 
+```
+03_decode_single_session.py X
+```
+where X is an integer which indexes across jobs.  Each sessions gets at least 1 job, so there are many 
```

Comment:
@bensonbrandon clarify that the pipeline is meant to be run across all sessions; cannot easily select a single session. point people to `decoding_example_script.py` for an example of how to decode with a single session.
Also add a new section at the end of the README that details how to run the pipeline on a smaller set of animals, say this can be helpful for debugging purposes.